parrot/src/pfs_table.cc

```diff
@@ -1479,7 +1513,12 @@ int pfs_table::symlink( const char *n1, const char *n2 )
 	verbatim down to the needed driver.
 	*/
 
-	if(resolve_name(0,n2,&pname,false)) {
+	path_dirname(n2,dirname2);
+
+	// Require write on the target to prevent linking into a RW directory
+	// and bypassing restrictions
+	if(resolve_name(0,n1,&pname,W_OK,false) &&
```

Comment:
I'd rather not have this check on the target of the symlink (`n1`). Correctly handling path resolution is the right way to fix this and that's a separate problem.